sklearn/neighbors/base.py

```diff
@@ -300,11 +300,26 @@ class from an array representing our data set and ask who's
         if self._fit_method is None:
             raise ValueError("must fit neighbors before querying")
 
-        X = check_array(X, accept_sparse='csr')
-
         if n_neighbors is None:
             n_neighbors = self.n_neighbors
 
+        query_is_train = True
+        if X is not None:
+            query_is_train = False
+            X = check_array(X, accept_sparse='csr')
+        else:
+            X = self._fit_X
+            # Include an extra neighbor in order to avoid marking the sample
+            # itself as the first neighbor.
+            if n_neighbors >= X.shape[0]:
+                raise ValueError(
+                    "The number of neighbors should be at least one less"
```

Comment:
This is confusing. How about "Expected n_neighbors <= {} for index of {} samples. Got {}." But why is a similar error not applicable to the `query_is_train=False` case?